api/v1alpha1/database_webhook.go

```diff
@@ -0,0 +1,164 @@
+
+var _ webhook.Validator = &Database{}
+
+func validateDatabaseCreate_NDBSpec(r *Database, allErrs field.ErrorList, ndbPath *field.Path) field.ErrorList {
+	databaselog.Info("Entering validateDatabaseCreate_NDBSpec...")
+	if r.Spec.NDB == (NDB{}) {
+		allErrs = append(allErrs, field.Invalid(ndbPath, r.Spec.NDB, "NDB spec must not be empty"))
+	}
+
+	if err := util.ValidateUUID(r.Spec.NDB.ClusterId); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("clusterId"), r.Spec.NDB.ClusterId, "ClusterId field must be a valid UUID"))
+	}
+
+	if r.Spec.NDB.CredentialSecret == "" {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("credentialSecret"), r.Spec.NDB.CredentialSecret, "CredentialSecret must not be empty"))
+	}
+
+	if err := util.ValidateURL(r.Spec.NDB.Server); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("server"), r.Spec.NDB.Server, "Server must be a valid URL"))
+	}
+
+	databaselog.Info("Exiting validateDatabaseCreate_NDBSpec...")
+	return allErrs
+}
+
+func validateDatabaseCreate_NewDBSpec(r *Database, allErrs field.ErrorList, instancePath *field.Path) field.ErrorList {
+	databaselog.Info("Entering validateDatabaseCreate_NewDBSpec...")
+
+	if r.Spec.Instance.DatabaseInstanceName == "" {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("databaseInstanceName"), r.Spec.Instance.DatabaseInstanceName, "Database Instance Name must not be empty"))
+	}
+
+	if len(r.Spec.Instance.DatabaseNames) < 1 {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("databaseNames"), r.Spec.Instance.DatabaseNames, "At least one Database Name must specified"))
+	}
+
+	if r.Spec.Instance.Size == 0 || r.Spec.Instance.Size < 10 {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("size"), r.Spec.Instance.Size, "Initial Database size must be 10 or more GBs"))
+	}
+
+	if r.Spec.Instance.CredentialSecret == "" {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("credentialSecret"), r.Spec.Instance.CredentialSecret, "CredentialSecret must not be empty"))
+	}
+
+	if _, isPresent := common.AllowedDatabaseTypes[r.Spec.Instance.Type]; !isPresent {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("type"), r.Spec.Instance.CredentialSecret, "Type must not be empty"))
+	}
+
+	databaselog.Info("Exiting validateDatabaseCreate_NewDBSpec...")
+	return allErrs
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Database) ValidateCreate() error {
+	databaselog.Info("Entering ValidateCreate...")
+
+	ndbSpecErrors := validateDatabaseCreate_NDBSpec(r, field.ErrorList{}, field.NewPath("spec").Child("ndb"))
+	dbSpecErrors := validateDatabaseCreate_NewDBSpec(r, field.ErrorList{}, field.NewPath("spec").Child("instance"))
+
+	allErrs := append(ndbSpecErrors, dbSpecErrors...)
+
+	combined_err := util.CombineFieldErrors(allErrs)
+	databaselog.Info("validate create database webhook response...", "combined_err", combined_err)
+
```

Comment:
Should we only log these errors when there are errors? Thinking about a len check of allErrs in an `if` block that contains line 143-144

Reply:
If there are no errors, let it print empty list, that way we can log the exact set of responses we received from the webhooks. Since we are not storing the logs in a file, it is just an extra stdout log, probably not going to be as bad.